lib/reply.js

```diff
@@ -56,7 +56,7 @@ Reply.prototype.send = function (payload) {
     this.res.setHeader('Content-Type', 'application/json')
 
     // Here we are assuming that the custom serializer is a json serializer
-    const str = this._serializer ? this._serializer(payload) : serialize(this.handle, payload)
+    const str = (this._serializer ? this._serializer(payload) : serialize(this.handle, payload)) || ''
```

Comment:
With empty content the code should be 204, we are already supporting that [here](https://github.com/fastify/fastify/blob/master/lib/reply.js#L44-L46), but we are not handling it correctly (in other words, we are not closing the response).
I think we should handle it all there and not here. What do you think?